src/sentry/api/serializers/rest_framework/notification_action.py

```diff
@@ -162,6 +163,9 @@ def validate_integration_and_service(self, data: NotificationActionInputData):
                     "integration_id": f"Integration of provider '{integration.provider}' does not match service type of '{service_provider}'"
                 }
             )
+        assert (
+            integration
+        ), f"Service type of '{service_provider}' requires having an active integration"
```

Comment:
maybe even above line 160 could be:

```python
if integration is None:
   raise serializers.ValidationError(...)
```

(and then it could remove the `integration and` part of the if statement after it)